src/main/java/de/tum/cit/aet/artemis/core/service/cleanup/DataCleanupService.java

```diff
@@ -0,0 +1,155 @@
+
+    private final ResultRepository resultRepository;
+
+    private final RatingRepository ratingRepository;
+
+    private final FeedbackRepository feedbackRepository;
+
+    private final TextBlockRepository textBlockRepository;
+
+    private final LongFeedbackTextRepository longFeedbackTextRepository;
+
+    private final StudentScoreRepository studentScoreRepository;
+
+    private final TeamScoreRepository teamScoreRepository;
+
+    private final ParticipantScoreRepository participantScoreRepository;
+
+    public DataCleanupService(CleanupJobExecutionRepository cleanupJobExecutionRepository, PlagiarismComparisonRepository plagiarismComparisonRepository,
+            ResultRepository resultRepository, RatingRepository ratingRepository, FeedbackRepository feedbackRepository, TextBlockRepository textBlockRepository,
+            LongFeedbackTextRepository longFeedbackTextRepository, StudentScoreRepository studentScoreRepository, TeamScoreRepository teamScoreRepository,
+            ParticipantScoreRepository participantScoreRepository) {
+        this.resultRepository = resultRepository;
+        this.ratingRepository = ratingRepository;
+        this.feedbackRepository = feedbackRepository;
+        this.textBlockRepository = textBlockRepository;
+        this.longFeedbackTextRepository = longFeedbackTextRepository;
+        this.studentScoreRepository = studentScoreRepository;
+        this.teamScoreRepository = teamScoreRepository;
+        this.cleanupJobExecutionRepository = cleanupJobExecutionRepository;
+        this.plagiarismComparisonRepository = plagiarismComparisonRepository;
+        this.participantScoreRepository = participantScoreRepository;
+    }
+
+    /**
+     * Deletes orphaned entities that are no longer associated with valid results or participations.
+     * This includes feedback, text blocks, and scores that reference null results, participations, or submissions.
+     *
+     * @return a {@link CleanupServiceExecutionRecordDTO} representing the execution record of the cleanup job
+     */
+    public CleanupServiceExecutionRecordDTO deleteOrphans() {
+        this.longFeedbackTextRepository.deleteLongFeedbackTextForEmptyFeedback();
+        this.textBlockRepository.deleteTextBlockForEmptyFeedback();
+        this.feedbackRepository.deleteOrphanFeedback();
+        this.studentScoreRepository.deleteOrphanStudentScore();
+        this.teamScoreRepository.deleteOrphanTeamScore();
+        this.longFeedbackTextRepository.deleteLongFeedbackTextForOrphanResult();
+        this.textBlockRepository.deleteTextBlockForOrphanResults();
+        this.feedbackRepository.deleteFeedbackForOrphanResults();
+        this.ratingRepository.deleteOrphanRating();
+        this.resultRepository.deleteResultWithoutParticipationAndSubmission();
+        return CleanupServiceExecutionRecordDTO.of(this.createCleanupJobExecution(CleanupJobType.ORPHANS, null, null));
```

Comment:
```suggestion
    public CleanupServiceExecutionRecordDTO deleteOrphans() {
        this.resultRepository.deleteResultWithoutParticipationAndSubmission();

        this.longFeedbackTextRepository.deleteLongFeedbackTextForEmptyFeedback();
        this.textBlockRepository.deleteTextBlockForEmptyFeedback();
        this.feedbackRepository.deleteOrphanFeedback();
        this.studentScoreRepository.deleteOrphanStudentScore();
        this.teamScoreRepository.deleteOrphanTeamScore();
        
        return CleanupServiceExecutionRecordDTO.of(this.createCleanupJobExecution(CleanupJobType.ORPHANS, null, null));
```
Shouldn't this be enough? This way you save calls to the db and can also remove some queries